src/utils/utils_part.c

```diff
@@ -191,81 +191,26 @@ void ocf_part_set_valid(struct ocf_cache *cache, ocf_part_id_t id,
 	}
 }
 
-static inline uint32_t ocf_part_evict_size(struct ocf_request *req)
+uint32_t ocf_part_overflow_size(struct ocf_cache *cache, struct ocf_user_part *part)
 {
-	uint32_t needed_cache_lines, part_available, cache_lines_to_evict;
-	uint32_t part_occupancy, part_occupancy_debt;
-	struct ocf_user_part *target_part = &req->cache->user_parts[req->part_id];
-	uint32_t part_occupancy_limit =
-		ocf_part_get_max_size(req->cache, target_part);
-
-	needed_cache_lines = ocf_engine_repart_count(req) +
-		ocf_engine_unmapped_count(req);
-
-	part_occupancy = ocf_part_get_occupancy(target_part);
-
-	if (part_occupancy_limit >= part_occupancy) {
-		part_available = part_occupancy_limit - part_occupancy;
-		part_occupancy_debt = 0;
-	} else {
-		/* Occupancy is greater than occupancy limit. Evict missing number of
-		 * cachelines, but no more than single eviction limit */
-		part_occupancy_debt = min((uint32_t)OCF_PENDING_EVICTION_LIMIT,
-				part_occupancy - part_occupancy_limit);
-		part_available = 0;
-	}
-
-	if (ocf_freelist_num_free(req->cache->freelist) <
-			ocf_engine_unmapped_count(req)) {
-		/* Number of cachelines to insert greater than number of free
-		 * cachelines */
-		if (part_available >= needed_cache_lines) {
-			/* Cache is full, but target's part occupancy limit is not reached
-			   */
-			ocf_req_clear_part_evict(req);
-			cache_lines_to_evict = needed_cache_lines;
-		} else {
-			/* Cache is full and target part reached it's occupancy limit */
-			ocf_req_set_part_evict(req);
-			cache_lines_to_evict = needed_cache_lines - part_available;
-		}
-
-	} else if (part_available < needed_cache_lines) {
-		/* Enough of free cache lines, but partition reached it's occupancy
-		 * limit */
-		cache_lines_to_evict = needed_cache_lines - part_available;
-		ocf_req_set_part_evict(req);
+	uint32_t part_occupancy = ocf_part_get_occupancy(part);
+	uint32_t part_occupancy_limit = ocf_part_get_max_size(cache, part);
 
-	} else if (part_available >= needed_cache_lines) {
-		/* Enough free cachelines available and they can be assigned to target
-		 * partition */
-		cache_lines_to_evict = 0;
+	if (part_occupancy > part_occupancy_limit)
+		return part_occupancy - part_occupancy_limit;
 
-	}
-
-	return cache_lines_to_evict + part_occupancy_debt;
+	return 0;
 }
 
-uint32_t ocf_part_check_space(struct ocf_request *req, uint32_t *to_evict)
+bool ocf_part_has_space(struct ocf_request *req)
```

Comment:
Inline?

Reply:
Since it's `ocf_part_has_space` maybe change the signature argument to `struct ocf_user_part` and add separate `target_partition(req)` macro for convenience?

Reply:
ok

Reply:
ok